### PR TITLE
Fix is statment for zero

### PR DESCRIPTION
**Pull request by hellysmile**

### CODE REVIEW

File: trafaret/__init__.py [hunk 1]
```diff
@@ -605,7 +605,7 @@ def __init__(self, allow_blank=False, regex=None, min_length=None, max_length=No
     def check_and_return(self, value):
         if not isinstance(value, str_types):
             self._failure("value is not a string", value=value)
-        if not self.allow_blank and len(value) is 0:
+        if not self.allow_blank and len(value) == 0:
```
Comment (bmwant) [inline trafaret/__init__.py:608]: I think it's ok to have such comparison, it's valid and will work even if it looks not logically right. Interpreter internally caches those numbers, so it's actually the same object in memory therefore `is` works fine

**Comment by bmwant** [comment]
@hellysmile Do you have an evidence where it's not working? If so can you also write a test to cover this codepath? And see my comment above

**Comment by Deepwalker** [comment]
@bmwant I think @hellysmile is right — it is interpereter dependent. Some alternative python versions can avoid this optimization.

**Comment by asvetlov** [comment]
Canonical version should be `if not self.allow_blank and not value:` :)

**Pull request merged**